.github/workflows/release.yml

```diff
@@ -0,0 +1,60 @@
+name: Build, Test, and Release on macOS and Windows
+
+on:
+  push:
+    branches:
+      - main  # Trigger on merge to the main branch
```

Comment:
You should temporarily add the current branch to this list for testing. Where will these releases go?